gobblin-modules/gobblin-azkaban/src/main/java/org/apache/gobblin/service/modules/orchestration/AzkabanClient.java

```diff
@@ -0,0 +1,463 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.TrustStrategy;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+
+/**
+ * A simple client that uses Ajax API to communicate with Azkaban server.
+ *
+ * Please note that this class can only be instantiated via {@link AzkabanClientBuilder}.
+ *
+ * @see {@linktourl https://azkaban.github.io/azkaban/docs/latest/#ajax-api}
+ */
+public class AzkabanClient implements Closeable {
+  protected final String username;
+  protected final String url;
+  protected String password;
+  protected String sessionId;
+  protected long sessionExpireInMin; // default value is 12h.
+  protected long sessionCreationTime = 0;
+  protected CloseableHttpClient client;
+  private static Logger log = LoggerFactory.getLogger(AzkabanClient.class);
+
+  protected AzkabanClient(AzkabanClientBuilder builder) throws AzkabanClientException {
+    this.url = builder.getUrl();
+    this.username = builder.getUsername();
+    this.password = builder.getPassword();
+    this.sessionExpireInMin = builder.getSessionExpireInMin();
+    this.client = getClient();
+    this.initializeSession();
+  }
+
+  /**
+   * Create a session id that can be used in the future to communicate with Azkaban server.
+   */
+  protected void initializeSession() throws AzkabanClientException {
+    try {
+      HttpPost httpPost = new HttpPost(this.url);
+      List<NameValuePair> nvps = new ArrayList<>();
+      nvps.add(new BasicNameValuePair(AzkabanClientParams.ACTION, "login"));
+      nvps.add(new BasicNameValuePair(AzkabanClientParams.USERNAME, this.username));
+      nvps.add(new BasicNameValuePair(AzkabanClientParams.PASSWORD, this.password));
+      httpPost.setEntity(new UrlEncodedFormEntity(nvps));
+      CloseableHttpResponse response = this.client.execute(httpPost);
+
+      try {
+        HttpEntity entity = response.getEntity();
+
+        // retrieve session id from entity
+        String jsonResponseString = IOUtils.toString(entity.getContent(), "UTF-8");
+        this.sessionId = parseResponse(jsonResponseString).get(AzkabanClientParams.SESSION_ID);
+        log.info("Session id : {}", this.sessionId);
```

Comment:
Remove this log to avoid any security issues.

Reply:
Done.